src/metabase/query_processor/middleware/annotate.clj

```diff
@@ -392,16 +369,38 @@
            :source :fields)))
 
 (mu/defn ^:private cols-for-ags-and-breakouts
```

Comment:
I'm a bit confused by this name. This seems to be doing more than returning the columns for aggregations and breakouts. It does care about columns coming from fields when setting the source, but the call to `cols-for-fields` below is still there.